specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1366 @@
+
+## 5.3. LocalSystem is not Supported
+
+Package dependencies can only be resolved to packages registered for a user. As packages cannot be
+registered for LocalSystem the Dynamic Dependencies feature is not available to callers running as LocalSystem.
+
+## 5.4. Packaging - ProjectReunion.Loader.dll
+
+The Dynamic Dependencies API is provided via ProjectReunion.Loader.dll.
+
+This dll will be included in the ProjectReunion Framework package for packaged applications.
+However, it must be a redistributable for non-packaged applications.
+
+Ironically, ProjectReunion.Loader.dll cannot be used from a Framework package by non-packaged applications.
+
+## 5.5. Dynamic Dependencies vis a vis Static Dependencies
+
+Dynamic dependencies' package dependency is equivalent to an MSIX package's `<PackageDependency>` in
+`appxmanifest.xml`. This defines criteria to be resolved to a package for use at runtime in a
+process' package graph.
+
+Packages dynamically added to a package graph can be removed. Packages in the package graph at
+process creation (aka static package graph) cannot be removed; the static package graph is a part of
+a process' DNA, from birth to death.
+
+## 5.6. Known Limitations in v1
+
+The initial implementation does not fully implement this spec.
+
+### 5.6.1. [uap6:LoaderSearchPathOverride](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-loadersearchpathoverride) not supported
+
+See [3.2.1. Known Issue: DLL Search Order for uap6:LoaderSearchPathOverride](#321-known-issue-dll-search-order-for-uap6loadersearchpathoverride).
+
+### 5.6.2. [uap6:AllowExecution](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-allowexecution) not supported
+
+See [3.2.2. Known Issue: DLL Search Order ignores uap6:AllowExecution](#322-known-issue-dll-search-order-ignores-uap6allowexecution).
+
+### 5.6.3. Lifecycle Hints are not supported
+
+A package dependency is pinned by a registered package. The lifecycle hints to unpin a package
+dependency when a file `LifecycleHint_FileOrPath`) or registry key (`LifecycleHint_RegistrySubkey`)
+is deleted are reserved for future use. Specifying these options today is an error.
+
+Implementing these options requires:
+
+1. Detecting when the lifetime artifact is deleted. This could be done aggressively (e.g. monitor the file/regkey) or lazily (e.g. check file/regkey existence when accessed)
+1. Knowing the package pinning the package dependency. This gets complicated if a pinning package declares 2+ `<PackageDependency>`.
+2. Calling `PackageManager.RemovePackageAsync(packagePinningThePackageDependency)`
+3. Sufficient rights to succeed
+
+# 6. API Details
+
+**TODO** Retitle this section 'API' since we don't have 'API Notes'?
+
+## 6.1. Win32 API - MsixDynamicDependency.hpp
+
+All Win32 APIs are prefixed with Mdd/MDD for MSIX Dynamic Dependencies.
+
+```c++
+enum class MddPinPackageDependency : uint32_t
```

Comment:
Maybe `MddPinPackageDependencyFlags` since that's what it is?

Reply:
We don't like flags named 'Flags'. Remember?

v0: `#define MDD_PIN_PACKAGEDEPENDENCY_FLAGS_LIFECYCLEHINT_PROCESS`
Feedback: Use enum, not #define. Even though a stable ABI export a C++ enum is OK (and preferred).

v1: `enum MddPinPackageDependencyFlags`
Feedback: Don't use 'Flags' in the name. Also, make it uint32_t.

v2: `enum MddPinPackageDependency : uint32_t`